js/components/address.js

```diff
@@ -0,0 +1,298 @@
+          }
+
+          if (types.includes('administrative_area_level_1')) {
+            addressData.state = component.long_name;
+          }
+
+          if (types.includes('country')) {
+            addressData.country = component.long_name;
+          }
+
+          if (types.includes('postal_code')) {
+            addressData.postalCode = component.long_name;
+          }
+        });
+      }
+
+      return addressData;
+    },
+    assignValuesToSeparateFields: function(place, separateFieldsName) {
+      var addressData = this.extractAddressComponents(place);
+
+      separateFieldsName.forEach(field => {
+        field.value = addressData[field.key];
+      });
+    },
+    updateSelectedFieldsProperty: function(attr, value) {
+      if (attr === 'readonly') {
+        value = !value;
+      }
+
+      var fields = this.$parent.fields;
+      var selectedValues = Object.values(this.selectedFieldOptions);
+
+      fields.forEach(field => {
+        if (selectedValues.includes(field.name)) {
+          field[attr] = value;
+        }
+      });
+    },
+    updateDisabledOptions: function() {
+      var selectedValues = Object.values(this.selectedFieldOptions);
+
+      this.fieldOptions.forEach(option => {
+        if (selectedValues.includes(option.label)) {
+          option.disabled = true;
+        } else {
+          option.disabled = false;
+        }
+      });
+    },
+    resetOptionsOnSelectOne: function(key) {
+      if (this.selectedFieldOptions[key] === '') {
+        this.updateDisabledOptions();
+      }
+    },
+    initAutocomplete: async function(input, countryRestrictions) {
+      this.addressField = Fliplet.UI.AddressField(this.$refs.addressField);
+      await this.addressField.getAutocompleteSuggestions(input, countryRestrictions)
+        .then(suggestions => {
+          if (typeof this.value === 'object') {
+            this.addressSuggestions = [];
+            this.suggestionSelected = true;
+          } else {
+            this.addressSuggestions = suggestions;
+          }
+        });
+    },
+    onChange: function() {
+      var $vm = this;
+
+      this.addressField.change((value) => {
+        if (this.addressComponents.length) {
+          this.assignValuesToSeparateFields(this.addressComponents, this.separateFieldsName);
+
+          for (const key in this.selectedFieldOptions) {
+            if (this.selectedFieldOptions[key]) {
+              var matchedField = this.separateFieldsName.find(field => field.key === key);
+              var fieldName = this.selectedFieldOptions[key];
+
+              if (matchedField) {
+                (function(fieldName, matchedFieldValue) {
+                  Fliplet.FormBuilder.get()
+                    .then(function(form) {
+                      form.field(fieldName).set(matchedFieldValue);
+                    });
+                })(fieldName, matchedField.value);
+              }
+            }
+          }
```

Comment:
@armine-fliplet, few improvements here:
1. use early returns to reduce nesting and cognitive complexity
2. There's no need for the inner self-calling function
```suggestion
            if (!this.selectedFieldOptions[key]) continue;
            
            const matchedField = this.separateFieldsName.find(field => field.key === key);
            const fieldName = this.selectedFieldOptions[key];

            if (!matchedField) continue;
              
            Fliplet.FormBuilder.get()
                .then(function(form) {
                  form.field(fieldName).set(matchedField.value);
                });
          }
```

Reply:
@dwfliplet Done